src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -821,18 +828,19 @@ def __init__(self, cli_ctx, auth_context_factory, adal_token_cache, arm_client_f
         def create_arm_client_factory(credentials):
             if arm_client_factory:
                 return arm_client_factory(credentials)
-            from azure.cli.core.profiles._shared import get_client_class
             from azure.cli.core.profiles import ResourceType, get_api_version
-            from azure.cli.core.commands.client_factory import configure_common_settings
-            from azure.cli.core.azclierror import CLIInternalError
-            client_type = get_client_class(ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS)
+            from azure.cli.core.commands.client_factory import _prepare_client_kwargs_track2
+
+            client_type = self._get_subscription_client_class()
             if client_type is None:
+                from azure.cli.core.azclierror import CLIInternalError
                 raise CLIInternalError("Unable to get '{}' in profile '{}'"
                                        .format(ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS, cli_ctx.cloud.profile))
             api_version = get_api_version(cli_ctx, ResourceType.MGMT_RESOURCE_SUBSCRIPTIONS)
+            client_kwargs = _prepare_client_kwargs_track2(cli_ctx)
+            # We don't need to change credential_scopes as 'scopes' is ignored by BasicTokenCredential anyway
             client = client_type(credentials, api_version=api_version,
-                                 base_url=self.cli_ctx.cloud.endpoints.resource_manager)
```

Comment:
What api-version of Subscription is used by azure-cli, the corresponding version of vendor SDK is required for azure-cli-core, right?

Reply:
Only `2019-11-01` and `2016-06-01` are used and they are included under `vendored_sdks`.

https://github.com/Azure/azure-cli/blob/0952571c2214693a79850f67d33b3fa6ab0354c0/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L155

https://github.com/Azure/azure-cli/blob/0952571c2214693a79850f67d33b3fa6ab0354c0/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L225

Reply:
I mean if a new api-version of Subscription is added to azure-cli in the feture, does the vendor SDK in azure-cli-core also need to change to this new api-version, right?

Reply:
`MGMT_RESOURCE_SUBSCRIPTIONS` is now only used by `azure-cli-core`. We can use it for now, assuming the corresponding `api-version` is vendored correctly.